integration-tests/src/test/java/com/datastax/oss/driver/api/core/ConnectIT.java

```diff
@@ -127,6 +133,36 @@ public void should_cleanup_on_lbp_init_failure() {
         .becomesTrue();
   }
 
+  /**
+   * Test for JAVA-2177. This ensures that even if the first attempted contact point is unreachable,
+   * its distance is set to LOCAL and reconnections are scheduled.
+   */
+  @Test
+  public void should_mark_unreachable_contact_points_as_local_and_schedule_reconnections() {
+    // Reject connections only on one node
+    BoundCluster boundCluster = simulacronRule.cluster();
+    boundCluster.node(0).rejectConnections(0, RejectScope.STOP);
+
+    try (CqlSession session = SessionUtils.newSession(simulacronRule)) {
+      Map<UUID, Node> nodes = session.getMetadata().getNodes();
+
+      // Before JAVA-2177, this would fail every other time because if the node was tried first for
+      // the initial connection, it was marked down and not passed to LBP.init(), and therefore
+      // stayed at distance IGNORED.
+      Node node0 = nodes.get(boundCluster.node(0).getHostId());
+      assertThat(node0.getState()).isEqualTo(NodeState.DOWN);
+      assertThat(node0.getDistance()).isEqualTo(NodeDistance.LOCAL);
+      assertThat(node0.getOpenConnections()).isEqualTo(0);
+      assertThat(node0.isReconnecting()).isTrue();
```

Comment:
should it be wrapped with the `ConditionChecker` and wait for it for some time? (I think it works in an async way)

Reply:
Good catch 👍 👍 👍 
If the "good" contact point is tried first, the "bad" one will remain UNKNOWN and we only detect it DOWN when we try to open the pool to it.
Opening the pool is not racy: the driver waits for all channels to fail, and triggers a reconnection. But the way the node state is updated is that that reconnection fires `ChannelEvent.reconnectionStarted(node)` when it starts. Channel events are processed by `NodeStateManager` on a separate thread (in this particular case we'll enter the branch `it has no connections and started reconnecting`).

So indeed we need to wait with a condition checker.